webui/react/src/pages/ExperimentDetails/ExperimentVisualization/LearningCurve.tsx

```diff
@@ -74,12 +77,58 @@ const LearningCurve: React.FC<Props> = ({
   const hasTrials = trialHps.length !== 0;
   const isExperimentTerminal = terminalRunStates.has(experiment.state as RunState);
 
-  const hyperparameters = useMemo(() => {
-    return fullHParams.reduce((acc, key) => {
-      acc[key] = experiment.hyperparameters[key];
+  const get_custom_search_varying_hps = (trialHps: TrialHParams[]) => {
+    /**
+     * For Custom Searchers, add a hyperparameter's column for params that
+     * 1) Have more than one unique value (it isn't the same in all trials)
+     * 2) Isn't a dictionary of other metrics
+     * This is to bypass the need to rely the on the experiment config's
+     * definition of hyperparameters and determine what should be shown more dynamically.
+     *
+     * Note: If we support the other tabs in the future for Custom Searchers
+     * such as HpParallelCoordinates, HpScatterPlots, and HpHeatMaps, we will need to
+     * generalize this logic a bit.
+     */
+    const uniq = trialHps.reduce((acc, d) => {
+      Object.keys(d.hparams).forEach((key: RecordKey) => {
+        const value = d.hparams[key];
+        if (!isNumber(value)) {
```

Comment:
I think this check will be an issue in general, since there will be important non-numerical hps to display.  For DS in particular, there are boolean hps that may differ across trials, but more generally there may be important strings, lists, etc.

Testing by modifying  `examples/custom_search_method/asha_search_method/remote_search_runner/run_experiment.py`
to
```python
def sample_params() -> Dict[str, object]:
    subdict = {
        "float": random.uniform(0.0, 1.0),
        "int": random.randint(0, 16),
        "str": random.choice("this is a string".split()),
        "list": [random.randint(0, n) for n in range(3)],
        "bool": random.choice([True, False]),
        "dict": {n: random.randint(0, n) for n in range(3)},
    }

    hparams = {
        "global_batch_size": 64,
        "n_filters1": random.randint(8, 64),
        "n_filters2": random.randint(8, 72),
        "learning_rate": 10 ** random.uniform(-4.0, 0.0),
        "dropout1": random.uniform(0.2, 0.8),
        "dropout2": random.uniform(0.2, 0.8),
        "subdict": subdict,
    }
    return hparams


```